client/ws-client/src/lib.rs

```diff
@@ -0,0 +1,179 @@
+///
+/// use jsonrpsee_ws_client::WsClientBuilder;
+///
+/// #[tokio::main]
+/// async fn main() {
+///     // build client
+///     let client = WsClientBuilder::default()
+///          .add_header("Any-Header-You-Like", "42")
+///          .build("wss://localhost:443")
+///          .await
+///          .unwrap();
+///
+///     // use client....
+/// }
+///
+/// ```
+#[derive(Clone, Debug)]
+pub struct WsClientBuilder<'a> {
+	certificate_store: CertificateStore,
+	max_request_body_size: u32,
+	request_timeout: Duration,
+	connection_timeout: Duration,
+	headers: Vec<Header<'a>>,
+	max_concurrent_requests: usize,
+	max_notifs_per_subscription: usize,
+	max_redirections: usize,
+}
+
+impl<'a> Default for WsClientBuilder<'a> {
+	fn default() -> Self {
+		Self {
+			certificate_store: CertificateStore::Native,
+			max_request_body_size: TEN_MB_SIZE_BYTES,
+			request_timeout: Duration::from_secs(60),
+			connection_timeout: Duration::from_secs(10),
+			headers: Vec::new(),
+			max_concurrent_requests: 256,
+			max_notifs_per_subscription: 1024,
+			max_redirections: 5,
+		}
+	}
+}
+
+impl<'a> WsClientBuilder<'a> {
+	/// Set whether to use system certificates
+	pub fn certificate_store(mut self, certificate_store: CertificateStore) -> Self {
+		self.certificate_store = certificate_store;
+		self
+	}
+
+	/// Set max request body size.
+	pub fn max_request_body_size(mut self, size: u32) -> Self {
+		self.max_request_body_size = size;
+		self
+	}
+
+	/// Set request timeout (default is 60 seconds).
+	pub fn request_timeout(mut self, timeout: Duration) -> Self {
+		self.request_timeout = timeout;
+		self
+	}
+
+	/// Set connection timeout for the handshake.
+	pub fn connection_timeout(mut self, timeout: Duration) -> Self {
+		self.connection_timeout = timeout;
+		self
+	}
+
+	/// Set a custom header passed to the server during the handshake.
+	///
+	/// The caller is responsible for checking that the headers do not conflict or are duplicated.
+	pub fn add_header(mut self, name: &'a str, value: &'a str) -> Self {
+		self.headers.push(Header { name, value: value.as_bytes() });
+		self
+	}
+
+	/// Set max concurrent requests.
+	pub fn max_concurrent_requests(mut self, max: usize) -> Self {
+		self.max_concurrent_requests = max;
+		self
+	}
+
+	/// Set max concurrent notification capacity for each subscription; when the capacity is exceeded the subscription
+	/// will be dropped.
+	///
+	/// You can also prevent the subscription being dropped by calling
+	/// [`Subscription::next()`](crate::types::Subscription) frequently enough such that the buffer capacity doesn't
+	/// exceeds.
+	///
+	/// **Note**: The actual capacity is `num_senders + max_subscription_capacity`
+	/// because it is passed to [`futures::channel::mpsc::channel`].
+	pub fn max_notifs_per_subscription(mut self, max: usize) -> Self {
+		self.max_notifs_per_subscription = max;
+		self
+	}
+
+	/// Set the max number of redirections to perform until a connection is regarded as failed.
```

Comment:
```suggestion
	/// Set the max number of redirects to follow during the WS handshake until a connection is considered failed.
```

Also: state the default in the docs.